### PR TITLE
Docker compat API - /images/search returns wrong structure (#7857)

### PR DESCRIPTION
**Pull request by mlegenovic**
Signed-off-by: Milivoje Legenovic <m.legenovic@gmail.com>

### CODE REVIEW

**Comment by mheon** [comment]
LGTM

File: pkg/api/handlers/compat/images_search.go [hunk 1]
```diff
@@ -77,5 +78,18 @@ func SearchImages(w http.ResponseWriter, r *http.Request) {
 		utils.BadRequest(w, "term", query.Term, err)
 		return
 	}
-	utils.WriteResponse(w, http.StatusOK, results)
+
+	compatResults := []registry.SearchResult{}
```
Comment (Luap99) [inline pkg/api/handlers/compat/images_search.go:82]:
```suggestion
	compatResults := make([]registry.SearchResult,0,len(results))
```
You should preallocate memory for the slice to reduce the memory copy operations.

**Comment by rhatdan** [comment]
LGTM with @Luap99 change.

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL
@Luap99 PTAL

**Pull request merged**